.github/workflows/release.yml

```diff
@@ -0,0 +1,41 @@
+name: do_release
+
+on:
+  push:
+    branches:
+     - 'master'
+    tags:
+      - '*'
+
+jobs:
+  goreleaser:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Code Checkout
+        uses: actions/checkout@v2
+      -
+        name: Run goreleaser
+        uses: ./.github/actions/releaser
```

Comment:
Just a dumb question for learning purposes as I'm not super familiar with docker. 

I assume this will run the `.github/workflows/releaser/action.yml` file, which says to run `.github/actions/releaser/Dockerfile`? This Dockerfile seems... light, how does it know what to do? Is it github actions that is smart enough on how to handle a docker file, run it, etc?

Reply:
ok so I see in the action that the job correctly runs `/usr/bin/docker run --name mailchaingoreleaserxcgolatest_c199d2 --label 179394 --workdir /github/workspace --rm -e GORELEASER_GITHUB_TOKEN -e BUILD_PLS_DIR -e INPUT_ARGS -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/stripe-cli/stripe-cli":"/github/workspace" mailchain/goreleaser-xcgo:latest --rm-dist`.

Where is that coming from? Is it part of the Dockerfile, or is it Github Action that is smart and knows to run this complicated command?

Reply:
GitHub actions knows to run the Dockerfile ([you can read more here](https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-docker-container-action)). The [base image has an entrypoint](https://github.com/mailchain/goreleaser-xcgo/blob/master/Dockerfile#L35) and because I omitted an entrypoint in my Dockerfile it will go ahead and run the base image's which is what we want. We'll need to start using a newer version of goreleaser for #549, so I proactively did the preparation work in this PR for that. IMO this is the most low touch and low maintenance way for us to keep using goreleaser-xcgo for binary cross compilation without getting stuck on old versions of goreleaser.

Reply:
> Where is that coming from?

It's mostly the GitHub Action runtime doing all of the work such as running the container and sharing all of the environment variables needed, including the ones we specify in our workflow file for that specific step.